pkg/reconciler/services/rabbit.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package services
+
+import (
+	"context"
+	"fmt"
+
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/eventing-rabbitmq/pkg/reconciler/broker/resources"
+	triggerresources "knative.dev/eventing-rabbitmq/pkg/reconciler/trigger/resources"
+	"knative.dev/eventing-rabbitmq/third_party/pkg/apis/rabbitmq.com/v1beta1"
+	rabbitclientset "knative.dev/eventing-rabbitmq/third_party/pkg/client/clientset/versioned"
+	rabbitmqclient "knative.dev/eventing-rabbitmq/third_party/pkg/client/injection/client"
+	bindinginformer "knative.dev/eventing-rabbitmq/third_party/pkg/client/injection/informers/rabbitmq.com/v1beta1/binding"
+	exchangeinformer "knative.dev/eventing-rabbitmq/third_party/pkg/client/injection/informers/rabbitmq.com/v1beta1/exchange"
+	queueinformer "knative.dev/eventing-rabbitmq/third_party/pkg/client/injection/informers/rabbitmq.com/v1beta1/queue"
+	rabbitlisters "knative.dev/eventing-rabbitmq/third_party/pkg/client/listers/rabbitmq.com/v1beta1"
+	"knative.dev/pkg/logging"
+)
+
+func NewRabbit(ctx context.Context) *Rabbit {
+	return &Rabbit{
+		rabbitClientSet: rabbitmqclient.Get(ctx),
+		exchangeLister:  exchangeinformer.Get(ctx).Lister(),
+		queueLister:     queueinformer.Get(ctx).Lister(),
+		bindingLister:   bindinginformer.Get(ctx).Lister(),
+	}
+}
+
+func NewRabbitTest(
```

Comment:
@benmoss There is a TODO here.. should we remove this before merging? Something for testing should be here right?